src/assets/wise5/classroomMonitor/dataExport/strategies/AbstractComponentDataExportStrategy.ts

```diff
@@ -0,0 +1,398 @@
+  }
+
+  private exportLatestWorkAndIsNotLatestWork(
+    componentStateIndex: number,
+    numComponentStates: number
+  ): boolean {
+    return (
+      this.workSelectionType === 'exportLatestWork' &&
+      componentStateIndex !== numComponentStates - 1
+    );
+  }
+
+  private incrementRevisionCounter(
+    componentRevisionCounter: ComponentRevisionCounter,
+    nodeId: string,
+    componentId: string
+  ): void {
+    componentRevisionCounter.incrementCounter(`${nodeId}_${componentId}`);
+  }
+
+  /**
+   * Create the array that will be used as a row in the student work export
+   * @param columnNames all the header column name
+   * @param columnNameToNumber the mapping from column name to column number
+   * @param rowCounter the current row number
+   * @param workgroupId the workgroup id
+   * @param userId1 the User ID 1
+   * @param userId2 the User ID 2
+   * @param userId3 the User ID 3
+   * @param periodName the period name
+   * @param componentRevisionCounter the mapping of component to revision counter
+   * @param componentState the component state
+   * @return an array containing the cells in the row
+   */
+  createStudentWorkExportRow(
+    columnNames: string[],
+    columnNameToNumber: Map<string, number>,
+    rowCounter: number,
+    workgroupId: number,
+    userId1: number,
+    userId2: number,
+    userId3: number,
+    studentName1: string,
+    studentName2: string,
+    studentName3: string,
+    periodName: string,
+    componentRevisionCounter: ComponentRevisionCounter,
+    componentState: any
+  ): string[] {
+    const row = new Array(columnNames.length);
+    row.fill('');
+    row[columnNameToNumber.get('#')] = rowCounter;
+    row[columnNameToNumber.get('Workgroup ID')] = workgroupId;
+    this.setStudentIDsAndNames(
+      row,
+      columnNameToNumber,
+      userId1,
+      studentName1,
+      userId2,
+      studentName2,
+      userId3,
+      studentName3
+    );
+    row[columnNameToNumber.get('Class Period')] = periodName;
+    row[columnNameToNumber.get('Project ID')] = this.configService.getProjectId();
+    row[columnNameToNumber.get('Project Name')] = this.projectService.getProjectTitle();
+    row[columnNameToNumber.get('Run ID')] = this.configService.getRunId();
+    row[columnNameToNumber.get('Student Work ID')] = componentState.id;
+    if (componentState.serverSaveTime != null) {
+      const formattedDateTime = this.utilService.convertMillisecondsToFormattedDateTime(
+        componentState.serverSaveTime
+      );
+      row[columnNameToNumber.get('Server Timestamp')] = formattedDateTime;
+    }
+    if (componentState.clientSaveTime != null) {
+      const clientSaveTime = new Date(componentState.clientSaveTime);
+      if (clientSaveTime != null) {
+        const clientSaveTimeString =
+          clientSaveTime.toDateString() + ' ' + clientSaveTime.toLocaleTimeString();
+        row[columnNameToNumber.get('Client Timestamp')] = clientSaveTimeString;
+      }
+    }
```

Comment:
Remove null-check? clientSaveTime and serverSaveTime are never null. I checked in prod db too.